qiskit/_quantumprogram.py

```diff
@@ -630,7 +630,7 @@ def get_initial_circuit(self):
     ###############################################################
 
     def set_api(self, token, url, hub=None, group=None, project=None,
-                proxies=None, verify=True):
+                proxies=None, verify=True, provider_name='ibmq'):
```

Comment:
`QuantumProgram` instances should be agnostic of providers and so, we should not provide a default value for the `provider_name` parameter.